aiogram_dialog/widgets/kbd/datecalendar.py

```diff
@@ -0,0 +1,134 @@
+            days = []
+            for w in monthcalendar(offset.year, offset.month):
+                week_row = []
+                for d in w:
+                    if d == 0:
+                        week_row.append(InlineKeyboardButton(text=" ",
+                                                             callback_data=" "))
+                    else:
+                        raw_data = int(mktime(date(offset.year, offset.month, d).timetuple()))
+                        week_row.append(InlineKeyboardButton(text=str(d),
+                                                             callback_data=f"{self.widget_id}:{raw_data}"))
+                days.append(week_row)
+
+            res = [
+                [
+                    InlineKeyboardButton(text=header_week,
+                                         callback_data=f"{self.widget_id}:zoom_one_year"),
+                ],
+                weekheader,
+                *days,
+                [
+                    InlineKeyboardButton(text="Prev month",
+                                         callback_data=f"{self.widget_id}:-"),
+                    InlineKeyboardButton(text="Zoom out",
+                                         callback_data=f"{self.widget_id}:zoom_one_year"),
+                    InlineKeyboardButton(text="Next month",
+                                         callback_data=f"{self.widget_id}:+"),
+                ],
+            ]
+
+        return res
+
+    async def process_callback(self, c: CallbackQuery, dialog: Dialog,
+                               manager: DialogManager) -> bool:
+        prefix = f"{self.widget_id}:"
+        offset = self.get_offset(manager)
+        if not c.data.startswith(prefix):
+            return False
+        data = c.data[len(prefix):]
+        if data == "+":
+            self.set_offset(datetime(offset.year + (offset.month // 12), ((offset.month % 12) + 1), 1), manager)
+        elif data == "-":
+            if offset.month == 1:
+                self.set_offset(datetime(offset.year - 1, 12, 1), manager)
+            else:
+                self.set_offset(datetime(offset.year, (offset.month - 1), 1), manager)
+        elif data == "zoom_one_year":
+            manager.context.set_data("outscope", "M", internal=True)
+        elif data == "zoom_years":
+            manager.context.set_data("outscope", "Y", internal=True)
+        elif "month" in data:
+            data = int(c.data[len(prefix) + 6:])
+            manager.context.set_data("outscope", False, internal=True)
+            self.set_offset(datetime(offset.year, data, 1), manager)
+        elif "year" in data:
+            data = int(c.data[len(prefix) + 5:])
+            manager.context.set_data("outscope", "M", internal=True)
+            self.set_offset(datetime(data, 1, 1), manager)
+        else:
+            raw_date = int(data)
+            manager.context.set_data(f"{self.widget_id}-selected", raw_date)
```

Comment:
`f"{self.widget_id}-selected"` can be really used as another widget id. Why not using one single key in context?